tests/Microsoft.Identity.Test.Unit.net45/CoreTests/WsTrustTests/WsTrustEndpointTests.cs

```diff
@@ -164,6 +165,39 @@ public void TestWsTrustEndpointUsernamePasswordWsTrust13()
             Assert.AreEqual(expectedMessage, requestMessage);
         }
 
+        [TestMethod]
+        [Description("Assert client info is included in userrealm discovery header")]
+        public void UserrealmDiscoveryHeadersTest()
+        {
+            string endpoint = "https://login.microsoftonline.com/common/userrealm/";
+
+            using (var harness = new MockHttpAndServiceBundle())
+            {
+                var requestContext = RequestContext.CreateForTest(harness.ServiceBundle);
+                var userrealmEndpoint = harness.ServiceBundle.WsTrustWebRequestManager.CreateUserrealmEndpoint(endpoint, _username, requestContext);
+
+                Assert.IsNotNull(userrealmEndpoint);
+
+                string[] pathSegments = userrealmEndpoint.Uri.AbsoluteUri.Substring(0).Split(
+                    new[]
+                    {
+                        '='
+                    });
+
+                string[] pathSegments2 = userrealmEndpoint.Uri.AbsoluteUri.Substring(0).Split(
+                    new[]
+                    {
+                        '&'
+                    },
+                    StringSplitOptions.RemoveEmptyEntries);
+                
+                Assert.AreEqual($"{endpoint}{_username}?api-version", pathSegments[0]);
+                StringAssert.Contains(pathSegments2[1], "x-client-SKU");
```

Comment:
>x-client-SKU" [](start = 57, length = 13)

consts from the product? I believe those are lower case typically?

Reply:
i pulled this from fiddler. if i put it in lower case the tests fail.

---
In reply to: [251518293](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/840#discussion_r251518293) [](ancestors = 251518293)

Reply:
the test should not fail. evo would not care. the test should compare ignoring case. should grab these values from constants imo

Reply:
done. thanks.